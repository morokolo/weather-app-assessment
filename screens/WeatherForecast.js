import React, { useEffect, useState } from "react";
import {
  ActivityIndicator,
  Platform,
  RefreshControl,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  View,
  StatusBar,
} from "react-native";
import userWeather from "../api/userWeather";
import CurrentWeather from "../components/CurrentWeather";
import Loader from "../components/Loader";
import WeeklyWeather from "../components/WeeklyWeather";
import bgColorDictionary from "../utils/bgColorDictionary";

export default function WeatherForecast() {
  const [refreshing, setRefreshing] = useState(false);
  const forecast = userWeather();

  if (!forecast) {
    return <Loader />;
  }
  const current = forecast.current.weather[0];
  return (
    <SafeAreaView style={styles.safeAreaContainer}>
      <ScrollView
        contentContainerStyle={styles.scrollViewContent}
        refreshControl={
          <RefreshControl
            onRefresh={() =>
              console.log("refactor the hooks properly in order to refresh")
            }
            refreshing={refreshing}
          />
        }
      >
        <View style={styles.weatherSectionContainer}>
          <CurrentWeather forecast={forecast} current={current} />
        </View>
        <View
          style={[
            styles.weatherSectionContainer,
            styles.weekly,
            {
              backgroundColor:
                bgColorDictionary[current.icon] || iconDictionary["02d"],
            },
          ]}
        >
          <WeeklyWeather forecast={forecast} />
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  scrollViewContent: {
    flex: 1,
    flexDirection: "column",
    marginTop: -25,
  },
  weatherSectionContainer: {
    flex: 1,
    marginTop: -55, //scroll
  },
  safeAreaContainer: {
    height: "100%",
    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
    marginTop: 0,
  },
  weekly: {
    flex: 1,
  },
});
