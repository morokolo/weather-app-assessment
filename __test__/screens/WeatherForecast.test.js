import "react-native";
import React from "react";
import WeatherForecast from "../../screens/WeatherForecast";
import renderer from "react-test-renderer";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("should render the WeatherForecast functional component", () => {
  const tree = renderer.create(<WeatherForecast />).toJSON();
  expect(tree).toMatchSnapshot();
});
