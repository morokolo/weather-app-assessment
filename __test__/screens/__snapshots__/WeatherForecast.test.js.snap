// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the WeatherForecast functional component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 0,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#fc5c65"
    hidesWhenStopped={true}
    size="large"
  />
</RCTSafeAreaView>
`;
